### ISSUE TITLE
Vocab produces non-deterministic vocab in py3.5

### ISSUE THREAD

**Issue by eric-haibin-lin** [dmlc/gluon-nlp#1158]
## Description
The special token IDs are registered in a non-deterministic order. Probably gluonnlp need some overhaul of all dict usages

**Comment by eric-haibin-lin** [comment]
Actually I suspect it impact all Vocab:
```
        for special_token_name, special_token in kwargs.items():
            # Test if kwarg specifies a special token
            if not special_token_name.endswith('_token'):
                raise ValueError('{} is invalid. Only keyword arguments '
                                 'that end in \'_token\' are supported '
                                 'to declare special tokens.'.format(special_token_name))

            if special_token is not None and special_token not in special_tokens:
                special_tokens.append(special_token)
```

**Comment by szha** [comment]
@eric-haibin-lin @leezu what option did you decide on?

**Comment by eric-haibin-lin** [comment]
I suggest we add the sort logic to py3.5 only and add a note for py3.5 users in the release note such that:
- python3.6+ users don't need to change anything
- for python3.5 users, they need to relay on serialized vocab in json format for compatibility if they use gluonnlp 0.9 and/or python3.6+

**Comment by eric-haibin-lin** [comment]
Fixed by #1166
